designsafe/apps/workspace/api/views.py

```diff
@@ -0,0 +1,293 @@
+                )
+            )
+            html_apps = list(
+                AppListingEntry.objects.filter(
+                    enabled=True,
+                    bundle__enabled=True,
+                    bundle__category=category,
+                    app_type="html",
+                )
+                .order_by(Coalesce("label", "app_id"))
+                .values(
+                    "app_id",
+                    "app_type",
+                    "bundle_id",
+                    "bundle_label",
+                    "html",
+                    "icon",
+                    "label",
+                    "license_type",
+                    "popular",
+                    "version",
+                )
+            )
+
+            valid_tapis_apps = self._get_valid_apps(apps_listing, tapis_apps)
+
+            categoryResult = {
+                "title": category.category,
+                "priority": category.priority,
+                "apps": [
+                    {k: v for k, v in app.items() if v != ""}
+                    for app in valid_tapis_apps
+                ]  # Remove empty strings from response
+                + html_apps,
+            }
+
+            categoryResult["apps"] = sorted(
+                categoryResult["apps"], key=lambda app: app["label"] or app["app_id"]
+            )
+            categories.append(categoryResult)
+
+        return categories
+
+    def get(self, request, *args, **kwargs):
+        """
+        Returns a structure containing app tray categories with the apps it contains
+
+        {
+            "categories": [
+                {
+                    "title": "Category 1",
+                    "priority": 1,
+                    "apps": [
+                        {
+                            "app_id": "jupyter-lab-hpc",
+                            "bundle_id": 1,
+                            "bundle_label": "Jupyter",
+                            "bundle_popular": True,
+                            "icon": "jupyter",
+                            "is_bundled": True,
+                            "label": Jupyter Lab HPC,
+                            "license_type": "Open Source",
+                            "version": "1.0.0",
+                            ...
+                        },
+                        ...
+                    ],
+                },
+                ...
+            ],
+        }
+        """
+        METRICS.info("user:%s op:%s" % (request.user.username, "getAppsTrayView"))
+
+        categories = self._getPublicApps(request.user)
+        my_apps = self._getPrivateApps(request.user)
+
+        categories.insert(0, {"title": "My Apps", "apps": my_apps})
+
+        # Only return categories that are non-empty
+        categories = list(
+            filter(lambda category: len(category["apps"]) > 0, categories)
+        )
+
+        return JsonResponse(
+            {"categories": categories}, encoder=BaseTapisResultSerializer
```

Comment:
What do you think about using Pydantic models to handle representation/serialization of the response here? Advantages would be: 
- provides a formal schema for the response in a more durable way than just documenting it
- automatic case conversion, so we can work in snake_case on the backend and camelCase on the frontend without the linter yelling at us on either side.

Reply:
I haven't used pydantic so hard to comment initially. Case conversion sounds useful, but would we be documenting the Tapis schema? If so that seems a bit heavier than just serializing the response

Reply:
Since we're combining Tapis and database results in these views, I would argue that we're already using a custom response schema, in a sense. Pydantic just makes that explicit. The schemas would be MUCH less involved than the ones we've been using in Projects, because they don't have heterogeneous legacy data to contend with. It would pretty much look like a dataclass with the fields we promise to return.

Reply:
We'll be returning entire tapipy objects in other calls though, with this same serializer. It sounds like we'd need to spec out the entire object